fastestimator/estimator.py

```diff
@@ -62,14 +62,40 @@ def __init__(self,
         self.monitor_names = to_set(monitor_names)
         self.system = System(log_steps=log_steps, epochs=epochs, max_steps_per_epoch=max_steps_per_epoch)
         self.trace_inputs = dict()
+        self.traces_in_use = []
 
     def fit(self):
         draw()
-        self.traces = self._prepare_traces()
+        self.traces_in_use = self._prepare_traces(modes=self.pipeline.get_modes())
         self._prepare_system()
-        self._check_keys()
+        self._check_keys(modes=self.pipeline.get_modes())
         self._warmup()
-        return self._start()
+        return self._start_train()
+
+    def test(self):
+        self.traces_in_use = self._prepare_traces(modes={"test"})
+        self._prepare_system()
+        self._check_keys(modes={"test"})
+        return self._start_test()
+
+    def _prepare_traces(self, modes: Set[str]):
+        self.trace_inputs = dict()
+        loss_keys = set()
+        traces = [trace for trace in self.traces]
+        if "train" in modes:
+            loss_keys = self.network.get_loss_keys()
+            traces.insert(0, TrainEssential())
+            if "eval" in modes:
```

Comment:
what if user wants to run 'eval' without 'train' to debug something?

Reply:
the differentiation of `eval` and `test` are:  eval happens during training, test does not. 

Therefore, if user wants to debug, put the data as test data then call estimator.test()

Reply:
there may be eval ops that they're trying to debug though, which wouldn't be caught be test mode. or are we saying that there is never a situation where network would be different for eval vs test?

Reply:
they can make steps_per_epoch to be 1, then run evall directly.         Running eval separately doesn't make sense.

Reply:
so in pipeline are we going to assert that if eval_data is present then train_data must be non None?

Reply:
we don't need to assert that in pipeline.    because pipeline can be independently used for non-training purpose as well.